contracts/contracts/vault/VaultAdmin.sol

```diff
@@ -103,6 +103,24 @@ contract VaultAdmin is VaultStorage {
         strategistAddr = _address;
     }
 
+    /**
+     * @dev Set the default Strategy for an asset, i.e. the one which the asset
+            will be automatically allocated to and withdrawn from
+     * @param _asset Address of the asset
+     * @param _strategy Address of the Strategy
+     */
+    function setAssetDefaultStrategy(address _asset, address _strategy)
+        external
+        onlyGovernorOrStrategist
+    {
+        emit AssetDefaultStrategyUpdated(_asset, _strategy);
```

Comment:
nit: Why is the event emitted at the start of the function just in this place?

Reply:
It removes slither warnings about reentrant events, some of the other functions required external calls before the event arguments were available so in those cases I generated a db for slither of ignored warnings in #386